File: ProyectoAsisTrack/src/main/java/com/example/ProyectoAsisTrack/Repository/HorariosRepository.java
package com.example.ProyectoAsisTrack.Repository;

import com.example.ProyectoAsisTrack.Model.Horarios;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface HorariosRepository extends JpaRepository<Horarios, Long> {
}

File: app/src/main/java/com/example/proyectoasistrack/data/Repository/HorariosRepository.kt
package com.example.proyectoasistrack.data.repository

import com.example.proyectoasistrack.data.dao.HorarioDao
import com.example.proyectoasistrack.data.model.Horario
import kotlinx.coroutines.flow.Flow

class HorariosRepository(private val horarioDao: HorarioDao) {

    fun getAllHorarios(): Flow<List<Horario>> = horarioDao.getAllHorarios()

    suspend fun insertHorario(horario: Horario) = horarioDao.insertHorario(horario)

    suspend fun updateHorario(horario: Horario) = horarioDao.updateHorario(horario)

    suspend fun deleteHorario(horario: Horario) = horarioDao.deleteHorario(horario)
}

